vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -208,20 +207,19 @@ fire_and_forget WinRTWebSocketResource::PerformWrite() noexcept
     }
 
     size_t length;
-    std::pair<string, bool> front;
-    bool popped = self->m_writeQueue.try_pop(front);
-    if (!popped)
+    string messageLocal;
+    bool isBinaryLocal;
     {
-      throw hresult_error(E_FAIL, L"Could not retrieve outgoing message.");
+      auto guard = lock_guard<mutex>{m_writeQueueMutex};
+      std::tie(messageLocal, isBinaryLocal) = m_writeQueue.front();
```

Comment:
>std::tie(messageLocal, isBinaryLocal) = m_writeQueue.front(); [](start = 6, length = 61)

Is this a common pattern? Looks overly complicated

Reply:
It’s a simple way to not rely on PPL containers.

This PR has been merged, but I can submit a new change if you know of a better way.